lib/Agrammon/UI/CommandLine.pm6

```diff
@@ -206,49 +246,91 @@ sub run (IO::Path $path, IO::Path $input-path, $technical-file, $variants, $form
 
     my $rc = Agrammon::ResultCollector.new;
     my atomicint $n = 0;
-    my %results;
     my class X::EarlyFinish is Exception {}
     race for $ds.read($fh).race(:$batch, :$degree) -> $dataset {
         my $my-n = ++⚛$n;
-
-        my $outputs = timed "$my-n: Run $filename", {
-            $model.run(
-                input     => $dataset,
-                technical => %technical-parameters,
-            );
+        my @validation-errors = validation-errors($model, $dataset);
+        if @validation-errors.elems {
```

Comment:
Don't need `.elems`, an array is truthy if it is non-empty. Could also do `if validation-errors($model, $dataset) -> @validation-errors { }` to more tightly scope it.

Reply:
Done